CMakeLists.txt

```diff
@@ -8,6 +8,7 @@ endif()
 if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 17)
 endif()
+set( CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -fPIC" )
```

Comment:
Don't set this globally. Instead, set it one by one on each target that needed it, e.g.:
```cmake
add_library(${PROJECT_NAME} SHARED
  ...
)

target_compile_options(${PROJECT_NAME} PRIVATE -fPIC)
```